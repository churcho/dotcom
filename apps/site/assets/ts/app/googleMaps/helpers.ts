// @ts-ignore
import stopMarker from "../../../static/images/icon-map-stop-marker.svg";
// @ts-ignore
import stationMarker from "../../../static/images/icon-map-station-marker.svg";
// @ts-ignore
import selectedStopMarker from "../../../static/images/icon-map-stop-marker-hover.svg";
// @ts-ignore
import selectedStationMarker from "../../../static/images/icon-map-station-marker-hover.svg";
// @ts-ignore
import currentLocationMarker from "../../../static/images/icon-current-location-marker.svg";
import { iconSize } from "../../../js/google-map/helpers";
import mapStyles from "../../../js/google-map/styles";
import { MarkerData, MapData } from "./__googleMaps";

export const stopIcon = (selected: boolean): string =>
  selected ? selectedStopMarker : stopMarker;
export const stationIcon = (selected: boolean): string =>
  selected ? selectedStationMarker : stationMarker;

export const buildIconFromSVG = (
  svg: string,
  size: string
): google.maps.Icon => {
  const sizeInt = iconSize(size);

  const encoded = window.btoa(svg);

  return {
    url: `data:image/svg+xml;base64, ${encoded}`,
    scaledSize: new window.google.maps.Size(sizeInt, sizeInt),
    origin: new window.google.maps.Point(0, 0),
    labelOrigin: new window.google.maps.Point(sizeInt / 2, sizeInt / 2)
  };
};

export const buildMarkerIcon = (
  data: MarkerData,
  selected: boolean
): google.maps.Icon | undefined => {
  if (data.icon) {
    if (data.icon!.includes("station")) {
      return buildIconFromSVG(stationIcon(selected), data.size);
    }
    if (data.icon!.includes("stop")) {
      return buildIconFromSVG(stopIcon(selected), data.size);
    }
    if (data.icon!.includes("current-location")) {
      return buildIconFromSVG(currentLocationMarker, data.size);
    }
  }
  return undefined;
};

export const setMapDefaults = (
  map: google.maps.Map,
  initialData: MapData
): void => {
  const options = {
    center: new window.google.maps.LatLng(
      initialData.default_center.latitude,
      initialData.default_center.longitude
    ),
    ...initialData.dynamic_options
  };
  const styles = mapStyles as google.maps.MapTypeStyle[];
  // Hide stop icons on the transit layer
  styles.push({
    featureType: "transit",
    elementType: "labels.icon",
    stylers: [{ visibility: "off" }]
  });
  map.setOptions({ styles, ...options });
  map.setZoom(initialData.zoom || 17);

  if (initialData.layers && initialData.layers.transit) {
    const layer = new window.google.maps.TransitLayer();
    layer!.setMap(map);
  }
};
