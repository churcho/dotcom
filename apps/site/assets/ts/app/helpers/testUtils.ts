// eslint-disable-next-line import/prefer-default-export
export const createReactRoot = (): void => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
};
