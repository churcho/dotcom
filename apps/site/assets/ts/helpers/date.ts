// Returns a date in the form of "August 7, 2019"
export const formattedDate = (unformatted: string): string => {
  const [year, month, day] = unformatted
    .split(/-/)
    .map(part => Number.parseInt(part, 10));

  // Remember that months in JS are 0-indexed for some reason, hence "month - 1".
  const parsedDate = new Date(year, month - 1, day);
  return parsedDate.toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC"
  });
};

// Returns a time in the form of "05:16 AM"
export const shortTime = (date: string): string =>
  new Date(date).toLocaleTimeString("en-US", {
    hour: "2-digit",
    minute: "2-digit"
  });

export default formattedDate;
