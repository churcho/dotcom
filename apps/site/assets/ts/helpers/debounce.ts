// Debounce is meant to be used with event listeners. Use it to delay a call
// until after a burst of events, such as keyboard input.
//
//   usage:
//
//     const callback = (ev: Event) => {}
//     document.addEventListener("keyup", debounce(callback, 250))
//

const debounce = (callback: Function, delay: number): EventListener => {
  let timer: number | undefined;

  return (...args) => {
    window.clearTimeout(timer);
    timer = window.setTimeout(() => callback(...args), delay);
  };
};

export default debounce;
