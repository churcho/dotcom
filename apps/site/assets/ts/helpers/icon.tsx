import renderSvg from "./render-svg";
// @ts-ignore
import accessibleIconSvg from "../../static/images/icon-accessible-default.svg";
// @ts-ignore
import blueLineIconSvg from "../../static/images/icon-blue-line-default.svg";
// @ts-ignore
import busIconSvg from "../../static/images/icon-mode-bus-default.svg";
// @ts-ignore
import commuterRailIconSvg from "../../static/images/icon-mode-commuter-rail-default.svg";
// @ts-ignore
import greenLineIconSvg from "../../static/images/icon-green-line-default.svg";
// @ts-ignore
import greenBLineIconSvg from "../../static/images/icon-green-line-b-default.svg";
// @ts-ignore
import greenCLineIconSvg from "../../static/images/icon-green-line-c-default.svg";
// @ts-ignore
import greenDLineIconSvg from "../../static/images/icon-green-line-d-default.svg";
// @ts-ignore
import greenELineIconSvg from "../../static/images/icon-green-line-e-default.svg";
// @ts-ignore
import mattapanLineIconSvg from "../../static/images/icon-mattapan-line-default.svg";
// @ts-ignore
import orangeLineIconSvg from "../../static/images/icon-orange-line-default.svg";
// @ts-ignore
import parkingIconSvg from "../../static/images/icon-parking-default.svg";
// @ts-ignore
import redLineIconSvg from "../../static/images/icon-red-line-default.svg";

export const accessibleIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, accessibleIconSvg);

export const blueLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, blueLineIconSvg);

export const busIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, busIconSvg);

export const commuterRailIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, commuterRailIconSvg);

export const greenLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, greenLineIconSvg);

export const greenBLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, greenBLineIconSvg);

export const greenCLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, greenCLineIconSvg);

export const greenDLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, greenDLineIconSvg);

export const greenELineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, greenELineIconSvg);

export const mattapanLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, mattapanLineIconSvg);

export const orangeLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, orangeLineIconSvg);

export const parkingIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, parkingIconSvg);

export const redLineIcon = (className: string = ""): JSX.Element =>
  renderSvg(className, redLineIconSvg);
