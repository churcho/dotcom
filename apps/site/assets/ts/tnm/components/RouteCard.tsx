/* eslint-disable react/prefer-stateless-function */
import React from "react";
import { StopCard, stopIsEmpty } from "./StopCard";
import { Route, SVGMarkers } from "./__tnm";

interface Props {
  route: Route;
  markers: SVGMarkers;
}

const routeIsEmpty = (route: Route): boolean => route.stops.every(stopIsEmpty);

const RouteCard = ({ route, markers }: Props) => {
  const bgClass = `u-bg--${routeBgColor(route)}`;

  if (routeIsEmpty(route)) {
    return null;
  }

  return (
    <div className="m-tnm-sidebar__route">
      <div className={`h3 m-tnm-sidebar__route-name ${bgClass}`}>
        <span className={busClass(route)}>{route.name}</span>
      </div>
      {route.stops.map(stop => (
        <StopCard key={stop.id} stop={stop} route={route} markers={markers} />
      ))}
    </div>
  );
};

export const isSilverLine = (route: Route): boolean => {
  const mapSet: { [routeId: string]: boolean } = {
    "741": true,
    "742": true,
    "743": true,
    "746": true,
    "749": true,
    "751": true
  };

  return mapSet[route.id] || false;
};

export const routeBgColor = (route: Route): string => {
  if (route.type === 2) return "commuter-rail";
  if (route.type === 4) return "ferry";
  if (route.id === "Red") return "red-line";
  if (route.id === "Orange") return "orange-line";
  if (route.id === "Blue") return "blue-line";
  if (route.id.startsWith("Green-")) return "green-line";
  if (isSilverLine(route)) return "silver-line";
  if (route.type === 3) return "bus";
  return "unknown";
};

export const busClass = (route: Route): string =>
  route.type === 3 && !isSilverLine(route) ? "bus-route-sign" : "";

export default RouteCard;
