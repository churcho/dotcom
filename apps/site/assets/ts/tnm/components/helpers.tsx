import React, { KeyboardEvent } from "react";

export const handleEnterKeyPress = (
  e: KeyboardEvent,
  onClick: Function
): void => {
  if (e.key === "Enter") {
    onClick();
  }
};

export const renderSvg = (className: string, svgText: string): JSX.Element => (
  <span
    className={className}
    // eslint-disable-next-line react/no-danger
    dangerouslySetInnerHTML={{ __html: svgText }}
  />
);
