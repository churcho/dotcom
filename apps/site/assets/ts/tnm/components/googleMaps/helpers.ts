// @ts-ignore
import stopMarker from "../../../../static/images/icon-map-stop-marker.svg";
// @ts-ignore
import stationMarker from "../../../../static/images/icon-map-station-marker.svg";
// @ts-ignore
import selectedStopMarker from "../../../../static/images/icon-map-stop-marker-hover.svg";
// @ts-ignore
import selectedStationMarker from "../../../../static/images/icon-map-station-marker-hover.svg";
// @ts-ignore
import currentLocationMarker from "../../../../static/images/icon-current-location-marker.svg";
import { iconSize } from "../../../../js/google-map/helpers";

import { MarkerData } from "../__tnm";

interface GoogleMapMarkerSVG {
  url: string;
  scaledSize: google.maps.Size;
  origin: google.maps.Point;
  anchor: google.maps.Point;
  labelOrigin: google.maps.Point;
}

export const stopIcon = (selected: boolean): string =>
  selected ? selectedStopMarker : stopMarker;
export const stationIcon = (selected: boolean): string =>
  selected ? selectedStationMarker : stationMarker;

export const buildIconFromSVG = (
  svg: string,
  size: string
): GoogleMapMarkerSVG => {
  const sizeInt = iconSize(size);

  const encoded = window.btoa(svg);

  return {
    url: `data:image/svg+xml;base64, ${encoded}`,
    scaledSize: new window.google.maps.Size(sizeInt, sizeInt),
    origin: new window.google.maps.Point(0, 0),
    anchor: new window.google.maps.Point(sizeInt / 2, sizeInt / 2),
    labelOrigin: new window.google.maps.Point(sizeInt / 2, sizeInt / 2)
  };
};

export const buildMarkerIcon = (
  data: MarkerData,
  selected: boolean
): GoogleMapMarkerSVG | undefined => {
  if (data.icon) {
    if (data.icon!.includes("station")) {
      return buildIconFromSVG(stationIcon(selected), data.size);
    }
    if (data.icon!.includes("stop")) {
      return buildIconFromSVG(stopIcon(selected), data.size);
    }
    if (data.icon!.includes("current-location")) {
      return buildIconFromSVG(currentLocationMarker, data.size);
    }
  }
  return undefined;
};
