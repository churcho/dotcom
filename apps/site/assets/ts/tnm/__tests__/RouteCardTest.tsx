import React from "react";
import renderer from "react-test-renderer";
import RouteCard, {
  isSilverLine,
  routeBgColor,
  busClass
} from "../components/RouteCard";
import { createReactRoot } from "./helpers/testUtils";
import {
  Route,
  Stop,
  TNMDirection,
  TNMHeadsign,
  TNMTime
} from "../components/__tnm";

/* eslint-disable typescript/camelcase */

const time: TNMTime = {
  scheduled_time: ["4:30", " ", "PM"],
  prediction: null
};

const headsign: TNMHeadsign = {
  name: "Headsign",
  times: [time],
  train_number: null
};

const direction: TNMDirection = {
  direction_id: 0,
  headsigns: [headsign]
};

const stop: Stop = {
  accessibility: ["wheelchair"],
  address: "123 Main St., Boston MA",
  closed_stop_info: null,
  "has_charlie_card_vendor?": false,
  "has_fare_machine?": false,
  id: "stop-id",
  "is_child?": false,
  latitude: 41.0,
  longitude: -71.0,
  name: "Stop Name",
  note: null,
  parking_lots: [],
  "station?": true,
  distance: "238 ft",
  directions: [direction],
  href: "/stops/stop-id"
};

const route: Route = {
  alert_count: 1,
  direction_destinations: ["Outbound Destination", "Inbound Destination"],
  direction_names: ["Outbound", "Inbound"],
  id: "route-id",
  name: "Route Name",
  header: "Route Header",
  long_name: "Route Long Name",
  description: "Route Description",
  stops: [stop],
  type: 3
};

it("it renders a stop card", () => {
  createReactRoot();
  const tree = renderer
    .create(<RouteCard route={route} dispatch={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("returns null if route has no schedules", () => {
  const stopWithoutSchedules: Stop = { ...stop, directions: [] };
  const routeWithoutSchedules: Route = {
    ...route,
    stops: [stopWithoutSchedules]
  };
  createReactRoot();
  const tree = renderer
    .create(<RouteCard route={routeWithoutSchedules} dispatch={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders a stop card for the silver line", () => {
  createReactRoot();
  const sl: Route = { ...route, id: "751" };
  const tree = renderer
    .create(<RouteCard route={sl} dispatch={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("isSilverLine", () => {
  it("identifies silver line routes", () => {
    ["741", "742", "743", "746", "749", "751"].forEach(id => {
      const sl: Route = { ...route, id };
      expect(isSilverLine(sl)).toBe(true);
    });
  });
});

describe("routeBgColor", () => {
  it("determines the background color by route", () => {
    const cr: Route = { ...route, type: 2 };
    expect(routeBgColor(cr)).toBe("commuter-rail");

    const ferry: Route = { ...route, type: 4 };
    expect(routeBgColor(ferry)).toBe("ferry");

    ["Red", "Orange", "Blue"].forEach(id => {
      const subway: Route = { ...route, type: 1, id };
      expect(routeBgColor(subway)).toBe(`${id.toLowerCase()}-line`);
    });

    const greenLine: Route = {
      ...route,
      type: 0,
      id: "Green-B"
    };
    expect(routeBgColor(greenLine)).toBe("green-line");

    const bus: Route = { ...route, type: 3, id: "1" };
    expect(routeBgColor(bus)).toBe("bus");

    const fake: Route = { ...route, type: 0, id: "fakeID" };
    expect(routeBgColor(fake)).toBe("unknown");
  });
});

describe("busClass", () => {
  it("determines a route is a bus route", () => {
    const bus: Route = { ...route, type: 3, id: "1" };
    expect(busClass(bus)).toBe("bus-route-sign");

    const notBus: Route = { ...route, type: 1, id: "Red" };
    expect(busClass(notBus)).toBe("");
  });
});
