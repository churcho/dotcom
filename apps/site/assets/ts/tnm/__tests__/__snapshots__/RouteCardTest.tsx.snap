// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a stop card 1`] = `
<div
  className="m-tnm-sidebar__route"
>
  <div
    className="h3 m-tnm-sidebar__route-name u-bg--bus"
  >
    <span
      className="bus-route-sign"
    >
      11
    </span>
  </div>
  <div
    className="m-tnm-sidebar__stop"
  >
    <div
      className="m-tnm-sidebar__stop-info"
    >
      <div
        className="m-tnm-sidebar__stop-name"
      >
        Washington St @ Essex St
      </div>
      <div
        className="m-tnm-sidebar__stop-distance"
      >
        0.2 mi
      </div>
    </div>
    <div>
      <div
        className="m-tnm-sidebar__direction"
      >
        <div
          className="m-tnm-sidebar__direction-name u-small-caps"
        >
          Inbound
        </div>
        <div
          className="m-tnm-sidebar__direction-destination"
        >
          Downtown Boston
        </div>
      </div>
      <div
        className="m-tnm-sidebar__headsign-schedule"
      >
        <div
          className="m-tnm-sidebar__headsign"
        >
          <div
            className="m-tnm-sidebar__headsign-name"
          >
            Downtown
          </div>
        </div>
        <div
          className="m-tnm-sidebar__schedules"
        >
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="m-tnm-sidebar__stop"
  >
    <div
      className="m-tnm-sidebar__stop-info"
    >
      <div
        className="m-tnm-sidebar__stop-name"
      >
        Tufts Medical Center
      </div>
      <div
        className="m-tnm-sidebar__stop-distance"
      >
        0.1 mi
      </div>
    </div>
    <div>
      <div
        className="m-tnm-sidebar__direction"
      >
        <div
          className="m-tnm-sidebar__direction-name u-small-caps"
        >
          Inbound
        </div>
        <div
          className="m-tnm-sidebar__direction-destination"
        >
          Downtown Boston
        </div>
      </div>
      <div
        className="m-tnm-sidebar__headsign-schedule"
      >
        <div
          className="m-tnm-sidebar__headsign"
        >
          <div
            className="m-tnm-sidebar__headsign-name"
          >
            Downtown
          </div>
        </div>
        <div
          className="m-tnm-sidebar__schedules"
        >
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a stop card for the silver line 1`] = `
<div
  className="m-tnm-sidebar__route"
>
  <div
    className="h3 m-tnm-sidebar__route-name u-bg--silver-line"
  >
    <span
      className=""
    >
      SL4
    </span>
  </div>
  <div
    className="m-tnm-sidebar__stop"
  >
    <div
      className="m-tnm-sidebar__stop-info"
    >
      <div
        className="m-tnm-sidebar__stop-name"
      >
        Washington St @ Essex St
      </div>
      <div
        className="m-tnm-sidebar__stop-distance"
      >
        0.2 mi
      </div>
    </div>
    <div>
      <div
        className="m-tnm-sidebar__direction"
      >
        <div
          className="m-tnm-sidebar__direction-name u-small-caps"
        >
          Inbound
        </div>
        <div
          className="m-tnm-sidebar__direction-destination"
        >
          South Station
        </div>
      </div>
      <div
        className="m-tnm-sidebar__headsign-schedule"
      >
        <div
          className="m-tnm-sidebar__headsign"
        >
          <div
            className="m-tnm-sidebar__headsign-name"
          >
            South Station
          </div>
        </div>
        <div
          className="m-tnm-sidebar__schedules"
        >
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="m-tnm-sidebar__stop"
  >
    <div
      className="m-tnm-sidebar__stop-info"
    >
      <div
        className="m-tnm-sidebar__stop-name"
      >
        Tufts Medical Center
      </div>
      <div
        className="m-tnm-sidebar__stop-distance"
      >
        0.1 mi
      </div>
    </div>
    <div>
      <div
        className="m-tnm-sidebar__direction"
      >
        <div
          className="m-tnm-sidebar__direction-name u-small-caps"
        >
          Outbound
        </div>
        <div
          className="m-tnm-sidebar__direction-destination"
        >
          Dudley
        </div>
      </div>
      <div
        className="m-tnm-sidebar__headsign-schedule"
      >
        <div
          className="m-tnm-sidebar__headsign"
        >
          <div
            className="m-tnm-sidebar__headsign-name"
          >
            Dudley
          </div>
        </div>
        <div
          className="m-tnm-sidebar__schedules"
        >
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                2
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                mins
              </div>
            </div>
          </div>
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                16
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                mins
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="m-tnm-sidebar__direction"
      >
        <div
          className="m-tnm-sidebar__direction-name u-small-caps"
        >
          Inbound
        </div>
        <div
          className="m-tnm-sidebar__direction-destination"
        >
          South Station
        </div>
      </div>
      <div
        className="m-tnm-sidebar__headsign-schedule"
      >
        <div
          className="m-tnm-sidebar__headsign"
        >
          <div
            className="m-tnm-sidebar__headsign-name"
          >
            South Station
          </div>
        </div>
        <div
          className="m-tnm-sidebar__schedules"
        >
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
          <div
            className="m-tnm-sidebar__schedule"
          >
            <div
              className="m-tnm-sidebar__time"
            >
              <div
                className="m-tnm-sidebar__time-number"
              >
                &lt; 1
              </div>
              <div
                className="m-tnm-sidebar__time-mins"
              >
                min
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`returns null if route has no schedules 1`] = `null`;
